### DISCUSSION TITLE
NodeKey reading from file

### DISCUSSION THREAD

**Discussion by Noc0r** [ConsenSys/quorum#1397] [Q&A]
Hi! I had researched the quorum code and found this
```
file: node/config.go
// NodeKey retrieves the currently configured private key of the node, checking
// first any manually set key, falling back to the one found in the configured
// data folder. If no key can be found, a new one is generated.
func (c *Config) NodeKey() *ecdsa.PrivateKey {
```
GetNodeKey() - call NodeKey() for getting node key

This code just reading secret key from file and use it for 
1. RegisterRaftService
2. Initializing P2P
3. engine = istanbulBackend.New(istanbulConfig, stack.GetNodeKey(), chainDb)
4.
```     
// force to set the istanbul etherbase to node key address
	if chainConfig.Istanbul != nil {
		eth.etherbase = crypto.PubkeyToAddress(stack.GetNodeKey().PublicKey)
	}
```
5. NewQuorumPermissionCtrl()

Why did you prefer to avoid KeyStore and keep private key in file? Does it a security issue? As far as i understand this code was only for testing purposes.
Hope you will make it clear for me :)

**Comment by antonydenyer** [comment]
It was probably done that way because that's what's available on the variable!

**Reply by Noc0r** [reply]
I got it, thank you for reply!
But it can lead to security issues, doesn't it? I mean, that private key located in datadir like plain text.
